lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Magazine.java

```diff
@@ -0,0 +1,23 @@
+package com.codedifferently.lesson10.library;
+
+import com.codedifferently.lesson10.library.exceptions.ItemCantBeCheckOutException;
+import com.codedifferently.lesson10.library.exceptions.LibraryNotSetException;
+
+/** Magazine */
+public class Magazine extends LibraryAsset {
+
+  public Magazine(String title, String isbn) {
+    this.title = title;
+    this.isbn = isbn;
+  }
+
+  @Override
+  public boolean isAbleToCheckOut() {
+    return false;
+  }
+
+  @Override
+  public boolean isCheckedOut() throws LibraryNotSetException {
+    throw new ItemCantBeCheckOutException("This Item Cant Be Checked Out.");
```

Comment:
Why are you throwing an exception here instead of just returning false? Does this really warrant blowing up?